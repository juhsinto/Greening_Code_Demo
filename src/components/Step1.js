import React from "react";

import {useForm} from "react-hook-form";
import { useDispatch, useSelector } from "react-redux";
import { useHistory } from "react-router-dom";
import { updatePage1, updateFormProgress } from "../rootSlice";
import {Result} from "./Result"
import {Header} from "./Header"

import {Jumbotron,Container,FormControl,InputGroup} from 'react-bootstrap';



export const Step1 = () => {
    const dispatch = useDispatch()
    const history = useHistory()
    const project_name = useSelector(state => state.project_name)
    const {register, handleSubmit, errors} = useForm({defaultValues: {project_name}});

    const onSubmit = (data) => {
        
        dispatch(
            updatePage1({
              ...data,
            })
          );
        dispatch(updateFormProgress(25));
        history.push("./step2")
    }

    

    return (
        <Container className="p-3">
            <Jumbotron>
                <Header />
                <form onSubmit={handleSubmit(onSubmit)}>
                    <div>
                        <label htmlFor="page1">Page 1 Details:</label>
                        <br />
                        
                        <InputGroup className="p-3">
                            <InputGroup.Prepend>
                              <InputGroup.Text id="inputGroup-sizing-default">Project Name</InputGroup.Text>
                            </InputGroup.Prepend>
                            <FormControl
                                type="text"
                                name="project_name" id="project_name" 
                                placeholder="Enter Project Name"
                                ref={
                                    register({
                                            required: "Project Name Required", 
                                    })
                                }
                            />
                        </InputGroup>
                        {errors.project_name && <p class="project_name" style={{color: "red"}}>{errors.project_name.message}</p>}

                        <InputGroup className="p-3">
                            <InputGroup.Prepend>
                              <InputGroup.Text id="inputGroup-sizing-default">Site Address</InputGroup.Text>
                            </InputGroup.Prepend>
                            <FormControl
                                type="text"
                                name="site_address" id="site_address" 
                                placeholder="Enter Site Address"
                                ref={
                                    register({
                                        required: "Site Address Required", 
                                        minLength: {value: 5, message: "Site Address Too short"} 
                                    })
                                }
                            />
                        </InputGroup>
                        {errors.site_address && <p class="site_address" style={{color: "red"}}>{errors.site_address.message}</p>}

                        <InputGroup className="p-3">
                            <InputGroup.Prepend>
                              <InputGroup.Text id="inputGroup-sizing-default">User Details</InputGroup.Text>
                            </InputGroup.Prepend>
                            <FormControl
                                type="text"
                                name="user_details" id="user_details" 
                                placeholder="Enter User Details (like Name)"
                                ref={
                                    register({
                                        required: "User Details Required", 
                                        minLength: {value: 1, message: "User Details Too short"} 
                                    })
                                }
                            />
                        </InputGroup>
                        {errors.user_details && <p class="user_details" style={{color: "red"}}>{errors.user_details.message}</p>}
                        
                        <br /><br />

                        <button type="submit">Next</button>
                        </div>
                </form>
            </Jumbotron>
            <Result />
        </Container>
    )
}

export default Step1;
