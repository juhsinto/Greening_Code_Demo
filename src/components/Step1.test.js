import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Step1 from './Step1';
import { store } from '../store';

it('renders Step1 correctly', () => {
  const tree = renderer.create(<Provider store={store}><Step1 /></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});