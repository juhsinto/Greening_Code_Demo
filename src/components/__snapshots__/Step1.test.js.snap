// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Step1 correctly 1`] = `
<div
  className="p-3 container"
>
  <div
    className="jumbotron"
  >
    <h1
      className="header"
    >
      A simple Questionnaire
    </h1>
    <form
      onSubmit={[Function]}
    >
      <div>
        <label
          htmlFor="page1"
        >
          Page 1 Details:
        </label>
        <br />
        <div
          className="p-3 input-group"
        >
          <div
            className="input-group-prepend"
          >
            <span
              className="input-group-text"
              id="inputGroup-sizing-default"
            >
              Project Name
            </span>
          </div>
          <input
            className="form-control"
            id="project_name"
            name="project_name"
            placeholder="Enter Project Name"
            type="text"
          />
        </div>
        <div
          className="p-3 input-group"
        >
          <div
            className="input-group-prepend"
          >
            <span
              className="input-group-text"
              id="inputGroup-sizing-default"
            >
              Site Address
            </span>
          </div>
          <input
            className="form-control"
            id="site_address"
            name="site_address"
            placeholder="Enter Site Address"
            type="text"
          />
        </div>
        <div
          className="p-3 input-group"
        >
          <div
            className="input-group-prepend"
          >
            <span
              className="input-group-text"
              id="inputGroup-sizing-default"
            >
              User Details
            </span>
          </div>
          <input
            className="form-control"
            id="user_details"
            name="user_details"
            placeholder="Enter User Details (like Name)"
            type="text"
          />
        </div>
        <br />
        <br />
        <button
          className="btn btn-primary"
          disabled={false}
          type="submit"
        >
          Next
        </button>
      </div>
    </form>
  </div>
  <p>
    JSON data:
  </p>
  <pre>
    {
  "page1": {},
  "page2": {},
  "page3": {},
  "formProgress": 0
}
  </pre>
  <p>
    Form Progress:
  </p>
  <div
    className="progress"
  >
    <div
      aria-valuemax={100}
      aria-valuemin={0}
      aria-valuenow={0}
      className="progress-bar"
      role="progressbar"
      style={
        Object {
          "width": "0%",
        }
      }
    >
      0%
    </div>
  </div>
</div>
`;
